dbos/_client.py

```diff
@@ -0,0 +1,211 @@
+        status = get_workflow(self._sys_db, self.workflow_id, True)
+        if status is None:
+            raise DBOSNonExistentWorkflowError(self.workflow_id)
+        return status
+
+
+class WorkflowHandleClientAsyncPolling(Generic[R]):
+
+    def __init__(self, workflow_id: str, sys_db: SystemDatabase):
+        self.workflow_id = workflow_id
+        self._sys_db = sys_db
+
+    def get_workflow_id(self) -> str:
+        return self.workflow_id
+
+    async def get_result(self) -> R:
+        res: R = await asyncio.to_thread(
+            self._sys_db.await_workflow_result, self.workflow_id
+        )
+        return res
+
+    async def get_status(self) -> "WorkflowStatus":
+        status = await asyncio.to_thread(
+            get_workflow, self._sys_db, self.workflow_id, True
+        )
+        if status is None:
+            raise DBOSNonExistentWorkflowError(self.workflow_id)
+        return status
+
+
+class DBOSClient:
+    def __init__(self, database_url: str, system_database: Optional[str] = None):
+        db_config = parse_database_url_to_dbconfig(database_url)
+        if system_database is not None:
+            db_config["sys_db_name"] = system_database
+        self._sys_db = SystemDatabase(db_config)
+
+    def destroy(self) -> None:
+        self._sys_db.destroy()
+
+    def _enqueue(
+        self, options: EnqueueOptions, *args: tuple[Any,], **kwargs: dict[str, Any]
+    ) -> str:
+        workflow_name = options["workflow_name"]
+        queue_name = options["queue_name"]
+
+        workflow_class_name = options.get("workflow_class_name")
+        app_version = options.get("app_version")
+        max_recovery_attempts = options.get("max_recovery_attempts")
+        if max_recovery_attempts is None:
+            max_recovery_attempts = DEFAULT_MAX_RECOVERY_ATTEMPTS
+        workflow_id = options.get("workflow_id")
+        if workflow_id is None:
+            workflow_id = str(uuid.uuid4())
+
+        status: WorkflowStatusInternal = {
+            "workflow_uuid": workflow_id,
+            "status": WorkflowStatusString.ENQUEUED.value,
+            "name": workflow_name,
+            "class_name": workflow_class_name,
+            "queue_name": queue_name,
+            "app_version": app_version,
+            "config_name": None,
+            "authenticated_user": None,
+            "assumed_role": None,
+            "authenticated_roles": None,
+            "request": None,
+            "output": None,
+            "error": None,
+            "created_at": None,
+            "updated_at": None,
+            "executor_id": None,
+            "recovery_attempts": None,
+            "app_id": None,
+        }
+
+        inputs: WorkflowInputs = {
+            "args": args,
+            "kwargs": kwargs,
+        }
+
+        wf_status = self._sys_db.insert_workflow_status(status)
+        self._sys_db.update_workflow_inputs(
+            workflow_id, _serialization.serialize_args(inputs)
+        )
```

Comment:
What happens if the status insert works but the update wf inputs throw?

Reply:
I think we had the same conversation in the TS DBOS Client.